codec-http2/src/main/java/io/netty/handler/codec/http2/StreamBufferingEncoder.java

```diff
@@ -210,17 +238,10 @@ private void tryCreatePendingStreams() {
         }
     }
 
-    private void cancelPendingStreams() {
-        Exception e = new Exception("Connection closed.");
-        while (!pendingStreams.isEmpty()) {
-            PendingStream stream = pendingStreams.pollFirstEntry().getValue();
-            stream.close(e);
-        }
-    }
-
     private void cancelGoAwayStreams(int lastStreamId, long errorCode, ByteBuf debugData) {
         Iterator<PendingStream> iter = pendingStreams.values().iterator();
-        Exception e = new GoAwayException(lastStreamId, errorCode, debugData);
+        ByteString goAwayData = new ByteString(ByteBufUtil.getBytes(debugData), false);
```

Comment:
not use the temporary var but just do it directly in the GoAwayException constructor ?